s3/src/test/scala/docs/scaladsl/S3SourceSpec.scala

```diff
@@ -243,6 +243,44 @@ class S3SourceSpec extends S3WireMockBase with S3ClientIntegrationSpec {
     result.futureValue.key shouldBe listKey
   }
 
+  it should "list keys and common prefixes for a given bucket with a prefix and delimiter" in {
+    mockListBucketAndCommonPrefixes()
+
+    //#list-bucket-and-common-prefixes
+    val keyAndCommonPrefixSource
+        : Source[(Seq[ListBucketResultContents], Seq[ListBucketResultCommonPrefixes]), NotUsed] =
+      S3.listBucketAndCommonPrefixes(bucket, listDelimiter, Some(listPrefix))
+    //#list-bucket-and-common-prefixes
+
+    val result = keyAndCommonPrefixSource.runWith(Sink.head)
+    val contents = result.futureValue._1
+    val commonPrefixes = result.futureValue._2
+
+    contents.head.key shouldBe listKey
+    commonPrefixes.head.prefix shouldBe listCommonPrefix
+  }
+
+  it should "list keys and common prefixes for a given bucket with a prefix and delimiter using the version 1 api" in {
+    mockListBucketAndCommonPrefixesVersion1()
+
+    //#list-bucket-and-common-prefixes-attributes
```

Comment:
I don't see a reference to this code snippet in docs.  These types of comments anchor code snippets that get extracted into paradox.

Reply:
true, that was c&p from the listBucket which is used for the S3 Attributes example. I guess it's still ok to test the passing of the attributes but no need for the paradox comments. I'll get rid of them